packages/flutter/lib/src/rendering/paragraph.dart

```diff
@@ -1335,6 +1335,14 @@ class _SelectableFragment with Selectable, ChangeNotifier implements TextLayoutM
       : paragraph._getOffsetForPosition(TextPosition(offset: selectionEnd));
     final bool flipHandles = isReversed != (TextDirection.rtl == paragraph.textDirection);
     final Matrix4 paragraphToFragmentTransform = getTransformToParagraph()..invert();
+    final TextSelection selection = TextSelection(
+      baseOffset: selectionStart,
+      extentOffset: selectionEnd,
+    );
+    final List<Rect> selectionRects = <Rect>[];
+    for (final TextBox textBox in paragraph.getBoxesForSelection(selection)) {
+      selectionRects.add(textBox.toRect());
```

Comment:
Do you mean if I click in the overlap between the selections?

<img width="421" alt="Screenshot 2023-06-15 at 11 28 44 AM" src="https://github.com/flutter/flutter/assets/948037/bd498134-af7c-45a1-9b3c-69197f72ebbc">

Reply:
I mean if it is possible the line height is large that each line does not overlap. I am not sure if that is possible though.

Reply:
I think it is possible given the current implementation.

<img width="421" alt="Screenshot 2023-06-15 at 12 02 51 PM" src="https://github.com/flutter/flutter/assets/948037/379db46c-b8df-4a24-8217-fb8f41585a82">

This is due to `getBoxesForSelection` defaulting to `SelectionHeightStyle.tight`.

```dart
  List<ui.TextBox> getBoxesForSelection(
    TextSelection selection, {
    ui.BoxHeightStyle boxHeightStyle = ui.BoxHeightStyle.tight,
    ui.BoxWidthStyle boxWidthStyle = ui.BoxWidthStyle.tight,
  }) {
    assert(!debugNeedsLayout);
    _layoutTextWithConstraints(constraints);
    return _textPainter.getBoxesForSelection(
      selection,
      boxHeightStyle: boxHeightStyle,
      boxWidthStyle: boxWidthStyle,
    );
  }
```

In `SelectableText` and `TextField` these values are customizable. Setting to `SelectionHeightStyle.max` would result in this instead.

<img width="421" alt="Screenshot 2023-06-15 at 12 04 59 PM" src="https://github.com/flutter/flutter/assets/948037/1c8a200a-4598-4772-9207-d7461755613e">

What I see on WordPad for windows is that it looks like they use `SelectionHeightStyle.max`.

Reply:
>  I think if the line do not overlap it might be reasonable to expect that the gap is not within the selection?

I don't have an answer for this, since I am not sure if there is a native app that does SelectionHeightStyle.tight?

I am ok as-is, just want to make sure we are aware of this.